proj1/generator.cpp

```diff
@@ -30,17 +28,20 @@ int main(int argc, char **argv) {
     argsParsing::ParsingResult result =
         argsParsing::parseArguments(argc, argv, HELP, 6);
 
-    if (strlen(argv[optind + 4]) != result.k) {
+    string initCtx = argv[optind + 4];
+    if (initCtx.length() != result.k) {
         cerr << "ERROR: beginSequence length must be equal to k" << endl;
-        exit(4);
+        exit(3);
     }
-    string initCtx = argv[optind + 4];
 
-    if (argc - 1 != optind + 5) {
```

Comment:
If the users doesn't insert numChars it will cause a "missing parameters errors" so this is not needed

Reply:
no, but numChars is mandatory, since if not provided the program doesn't know qhen to stop

Reply:
On generator.cpp we call [parseArguments](https://github.com/joao-alegria/TAI/blob/d99b45f4de1dff379da037b5f6d8e87fa8bfc275/proj1/generator.cpp#L29) with the argument "numOfArgs" equal to 6, so If the user doesn't insert "numChars" [this](https://github.com/joao-alegria/TAI/blob/d99b45f4de1dff379da037b5f6d8e87fa8bfc275/proj1/argsParsing.cpp#L40) condition will exit the program since he only inserted 5

Reply:
that's correct, i didn't remember
then it's not necessary as you mentioned